hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetManagement.java

```diff
@@ -851,6 +855,36 @@ public boolean existsByControllerId(final String controllerId) {
         return targetRepository.exists(TargetSpecifications.hasControllerId(controllerId));
     }
 
+    @Override
+    public boolean isDeviceMatchingTargetFilterQueryAndNonDSAndCompatible(final String controllerId,
+            final long distributionSetId, final String targetFilterQuery) {
+        if (!isTargetFilterQueryValid(targetFilterQuery)) {
```

Comment:
I would rather call `RSQLUtility.validateRsqlFor` here directly without catching exception because if the query is not valid, I need to know it rather than just implicitly swallowing the problem

Reply:
done